### PR TITLE
[Draft][NO MERGE] Fix errors related with luci/profile

### PR DESCRIPTION
**Pull request by llFreetimell**
This draft is for fixing errors and discussing related with luci/profile.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

**Comment by llFreetimell** [comment]
More errors can be found. If then, they will be added here.

**Comment by llFreetimell** [comment]
@mhs4670go 

Let's suppose that `nodeA` has `origin1`.
Then `add_origin(nodeA, origin2)` is accepted as following options for now...
- `add_origin(nodeA, origin2)` = origin(nodeA) + origin2 = origin1 + origin2
- `add_origin(nodeA, origin2)` = origin2
I implemented `add_origin` as latter, but some of `luci/pass` are using like former.
We need to unify this...

**Comment by mhs4670go** [comment]
@llFreetimell Well, I thought you did it former. Then, how about changing the name to `set_origin` as you said before?

File: compiler/luci/profile/src/CircleNodeOrigin.cpp [hunk 1]
```diff
@@ -141,13 +141,22 @@ namespace luci
 
 bool has_origin(const luci::CircleNode *circle_node)
 {
-  return circle_node->annot<CircleNodeOriginAnnotation>() != nullptr;
+  if (circle_node->annot<CircleNodeOriginAnnotation>() == nullptr)
+    return false;
+  if (circle_node->annot<CircleNodeOriginAnnotation>()->origin() == nullptr)
+    return false;
```
Comment (mhs4670go) [inline compiler/luci/profile/src/CircleNodeOrigin.cpp:147]: Maybe errors wouldn't occur without this line, would they?

File: compiler/luci/profile/src/CircleNodeOrigin.cpp [hunk 2]
```diff
@@ -141,13 +141,22 @@ namespace luci
...
+  return true;
 }
 
 void add_origin(luci::CircleNode *circle_node, const std::shared_ptr<CircleNodeOrigin> origin)
 {
+  if (origin == nullptr)
+    return;
+
+  auto new_origin = composite_origin({get_origin(circle_node), origin});
```
Comment (mhs4670go) [inline compiler/luci/profile/src/CircleNodeOrigin.cpp:156]: This is not the way you said in the comments. Is it a former?

File: compiler/luci/profile/src/CircleNodeOrigin.cpp (hunk 1 above)
Comment (llFreetimell) [reply compiler/luci/profile/src/CircleNodeOrigin.cpp:147]: Ah, I added this for another purpose but not yet removed :(
ignore this :)

File: compiler/luci/profile/src/CircleNodeOrigin.cpp (hunk 2 above)
Comment (llFreetimell) [reply compiler/luci/profile/src/CircleNodeOrigin.cpp:156]: This draft is for discussing and recording where to fix. So the code is not fixed :)

**Comment by llFreetimell** [comment]
Related PR will be created soon, close!

**Pull request closed without merging**